tools/integration_tests/emulator_tests/write_stalls/chunk_transfer_timeout_infinity_test.go

```diff
@@ -0,0 +1,116 @@
+
+package emulator_tests
+
+import (
+	"crypto/rand"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path"
+	"testing"
+	"time"
+
+	emulator_tests "github.com/googlecloudplatform/gcsfuse/v2/tools/integration_tests/emulator_tests/util"
+	"github.com/googlecloudplatform/gcsfuse/v2/tools/integration_tests/util/setup"
+	"github.com/googlecloudplatform/gcsfuse/v2/tools/integration_tests/util/test_setup"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+////////////////////////////////////////////////////////////////////////
+// Boilerplate
+////////////////////////////////////////////////////////////////////////
+
+const (
+	fileSize  = 50 * 1024 * 1024
+	stallTime = 40 * time.Second
+)
+
+var configPath = "../proxy_server/configs/write_stall_40s.yaml"
+
+type chunkTransferTimeoutInfinity struct {
+	flags []string
+}
+
+func (s *chunkTransferTimeoutInfinity) Setup(t *testing.T) {
+	emulator_tests.StartProxyServer(configPath)
+	setup.MountGCSFuseWithGivenMountFunc(s.flags, mountFunc)
+}
+
+func (s *chunkTransferTimeoutInfinity) Teardown(t *testing.T) {
+	setup.UnmountGCSFuse(rootDir)
+	assert.NoError(t, emulator_tests.KillProxyServerProcess(port))
+}
+
+////////////////////////////////////////////////////////////////////////
+// Test scenarios
+////////////////////////////////////////////////////////////////////////
+
+// This test verifies that write operations stall for the expected duration
+// when write stall is induced while uploading first chunk.
+// It creates a file, writes data to it, and then calls Sync() to ensure
+// the data is written to GCS. The test measures the time taken for the Sync()
+// operation and asserts that it is greater than or equal to the configured stall time.
+func (s *chunkTransferTimeoutInfinity) TestWriteStallCausesDelay(t *testing.T) {
+	testDir := "TestWriteStallCausesDelay"
+	testDirPath = setup.SetupTestDirectory(testDir)
+	filePath := path.Join(testDirPath, "file.txt")
+	// Create a file for writing
+	file, err := os.Create(filePath)
+	if err != nil {
+		require.NoError(t, err)
+	}
+	defer file.Close()
+
+	// Generate random data
+	data := make([]byte, fileSize)
+	if _, err := io.ReadFull(rand.Reader, data); err != nil {
+		require.NoError(t, err)
+	}
+
+	// Write the data to the file
+	if _, err := file.Write(data); err != nil {
+		assert.NoError(t, err)
+	}
+	startTime := time.Now()
+	err = file.Sync()
+	endTime := time.Now()
+	elapsedTime := endTime.Sub(startTime)
+	assert.NoError(t, err)
+
+	assert.GreaterOrEqual(t, elapsedTime, stallTime)
+}
+
+////////////////////////////////////////////////////////////////////////
+// Test Function (Runs once before all tests)
+////////////////////////////////////////////////////////////////////////
+
+func TestChunkTransferTimeoutInfinity(t *testing.T) {
+	ts := &chunkTransferTimeoutInfinity{}
+	proxyEndpoint := fmt.Sprintf("http://localhost:%d/storage/v1/b?project=test-project/b?bucket=%s", port, setup.TestBucket())
```

Comment:
Why are we providing bucket-name as proxyEndpoint? Shouldn't it take as it is via GCSFuse mount option?

Reply:
Hmm makes sense updated. Although in my testing I used same type of URL and it was working fine.